File: Paseo2/src/main/java/modelos/RubroDeProducto.java
package modelos;

import java.util.List;
import java.util.ArrayList;
import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import utils.Estado;

@Entity
public class RubroDeProducto {
	
	@Id @GeneratedValue(strategy=GenerationType.IDENTITY) 
	private Long id;
	
    private String nombre;
    
    @Enumerated(EnumType.ORDINAL)
    private Estado estado;
    
    @ManyToMany(mappedBy ="rubros")
    @JsonIgnore
    private List<Producto> productos= new ArrayList<>();
    
    public RubroDeProducto() {
    	
    }
	public RubroDeProducto(String nombre, Estado estado, List<Producto> productos) {
		super();
		this.nombre = nombre;
		this.estado = estado;
		this.productos = productos;
	}

	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public Estado getEstado() {
		return estado;
	}
	public void setEstado(Estado estado) {
		this.estado = estado;
	}
	public List<Producto> getProductos() {
		return productos;
	}
	public void setProductos(List<Producto> productos) {
		this.productos = productos;
	}


	
	
    
}

File: PaseoFrontEnd/src/app/modelos/rubro-de-producto.ts
import { Estado } from "../utils/estado";
import { Producto } from "./producto";

export class RubroDeProducto {

     id:number;
     nombre:string;
     estado:Estado;
     productos:Producto[];


	constructor(nombre: string, estado: Estado, productos: Producto[]) {
		this.nombre = nombre;
		this.estado = estado;
		this.productos = productos;
	}


    /**
     * Getter id
     * @return {number}
     */
	public get Id(): number {
		return this.id;
	}

    /**
     * Getter nombre
     * @return {string}
     */
	public get Nombre(): string {
		return this.nombre;
	}

    /**
     * Getter estado
     * @return {Estado}
     */
	public get Estado(): Estado {
		return this.estado;
	}

    /**
     * Getter productos
     * @return {Producto[]}
     */
	public get Productos(): Producto[] {
		return this.productos;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set Id(value: number) {
		this.id = value;
	}

    /**
     * Setter nombre
     * @param {string} value
     */
	public set Nombre(value: string) {
		this.nombre = value;
	}

    /**
     * Setter estado
     * @param {Estado} value
     */
	public set Estado(value: Estado) {
		this.estado = value;
	}

    /**
     * Setter productos
     * @param {Producto[]} value
     */
	public set Productos(value: Producto[]) {
		this.productos = value;
	}

}
